docs/csharp/language-reference/builtin-types/enum.md

```diff
@@ -0,0 +1,92 @@
+---
+title: "Enumeration types - C# reference"
+description: "Learn about C# enumeration types that represent a set of named constants and how to define them with the `enum` keyword"
+ms.date: 09/12/2019
+f1_keywords:
+  - "enum"
+  - "enum_CSharpKeyword"
+helpviewer_keywords:
+  - "enum keyword [C#]"
+  - "enum type [C#]"
+  - "enumeration type [C#]"
+  - "bit flags [C#]"
+ms.assetid: bbeb9a0f-e9b3-41ab-b0a6-c41b1a08974c
+---
+# Enumeration types (C# reference)
+
+An enumeration type (or enum type) is a value type that represents a set of named integral constants. To define an enumeration type, use the `enum` keyword and specify the names of *enum members*:
+
+```csharp
+enum Season
+{
+    Spring,
+    Summer,
+    Autumn,
+    Winter
+}
+```
+
+An enumeration type has an underlying type, which can be any [integral numeric type](integral-numeric-types.md) and is `int` by default. Each enum member has an associated constant value of the underlying type. By default, if an enum member is not the first declared member, its associated value is the associated value of the preceding enum member increased by one; otherwise, it's zero. You can explicitly specify an underlying type or associated constant values, as the following example shows:
+
+```csharp
+enum ErrorCode : ushort
+{
+    None = 0,
+    Unknown = 1,
+    ConnectionLost = 100,
+    OutlierReading = 200
+}
+```
+
+The default value of an enumeration type `E` is the value produced by expression `(E)0`.
+
+> [!NOTE]
+> If you don't specify associated values of enum members, the first declared enum member has the associated value 0 and is the default value of an enumeration type.
+>
+> If you explicitly specify associated values of enum members and want to ensure that the default value of an enumeration type is a declared value, declare an enum member with the associated value 0.
+
+The <xref:System.Enum?displayProperty=nameWithType> type is the abstract base class of all enumeration types. It provides a number of methods to get information about an enumeration type and its values. For more information and examples, see the <xref:System.Enum?displayProperty=nameWithType> API reference page.
+
+Beginning with C# 7.3, you can use `System.Enum` in a base class constraint (that is known as the [enum constraint](../../programming-guide/generics/constraints-on-type-parameters.md#enum-constraints)) to specify that a type parameter is an enumeration type.
+
+You cannot define a method inside the definition of an enumeration type, but you can create an [extension method](../../programming-guide/classes-and-structs/extension-methods.md) to add functionality to an enumeration type.
+
+Like with any [constant](../../programming-guide/classes-and-structs/constants.md), at compile time all references to individual values of an enumeration type are replaced with the corresponding integral values.
```

Comment:
```suggestion
Like with any [constant](../../programming-guide/classes-and-structs/constants.md), at compile time, all references to individual values of an enumeration type are replaced with the corresponding integral values.
```

Reply:
Where can I find the guidance for those comma suggestions? I wonder if those commas are must or can be dropped (that is, both variants would be possible).

Reply:
I don't know if there is a guidance on that. but I think it reads better with the comma.

Reply:
@Youssef1313 I've tried to rephrase sentences to avoid such situations.